### ISSUE TITLE
[Term Entry] Python Neural Networks: Adagrad

### ISSUE THREAD

**Issue by codecademy-docs** [Codecademy/docs#6380]
### Reason/inspiration (optional)

We would like a new entry on the `Adagrad` term under Neural Networks. The entry should go in a new file under `docs/content/ai/concepts/neural-networks/terms/adagrad/adagrad.md`.

The entry should include:

- An introduction to the concept
- A section about Adagrad Optimizer works
- Difference between Adagrad, Adam, RMSprop
- An `Example` section that provides an example demonstrating its implementation with PyTorch
- 

Please refer to the [term entry template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md) and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [x] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [x] Labels added
- [ ] Issue is assigned
